Epub crash: probably caused by missing CSS

https://support.andbible.org/scp/tickets.php?id=3114

```
--- PLEASE WRITE YOUR FEEDBACK OR BUG REPORT ABOVE THIS LINE ---

Instructions:
Tell us briefly what did you do / what happened when issue took place.

For example:
  1. Opened app
  2. Clicked window menu
  -> Crash
 
Please use this function to report as soon as you have noticed issue taking place. You can also use this function after opening app, after crash has taken place.

Attachments:
  - logcat.txt.gz: Log file produced by this app. It does not contain information about any other apps.
  - screenshot.webp: Screenshot of app when crash took place, or when bug report was created.
  - last_crash_logcat.txt.gz: Logcat from the last app crash
  - last_crash_screenshot.webp: Screenshot from the last app crash

Having these files attached helps app developers a lot in fixing the reported bug. Sometimes manually taken screenshots also are beneficial in resolving issues.

Device info:
App id: net.bible.android.activity
Version: 5.0.902#83335ce89 fdroid standard release (27/11/25 15:20:40)
Android version: 12
Android SDK version: 31
Manufacturer: samsung
Hardware: qcom
Product: m11qnnxx
Device: m11q
Brand: samsung
Model: SM-M115F
Storage Mb free: 1037
WebView version: 143.0.7499.34
SQLITE version: 3.49.0
Used heap memory in Mb: 14
Max heap memory in Mb: 192

Current Exception:
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:581)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
... 1 more
Caused by: java.io.FileNotFoundException: /storage/emulated/0/Android/data/net.bible.android.activity/files/epub/Publishers, Tyndale House - Daily Walk Bible-NLT (1997, Tyndale House Publishers) - libgen.li.epub/OEBPS/Text/XXXXXXXXXXXXXXXX: open failed: ENOENT (No such file or directory)
at libcore.io.IoBridge.open(IoBridge.java:574)
at java.io.FileInputStream.(FileInputStream.java:160)
at kotlin.io.FilesKt__FileReadWriteKt.readBytes(FileReadWrite.kt:69)
at net.bible.android.view.activity.page.BibleView$ModuleStylesAssetHandler.handle$lambda$0(BibleView.kt:939)
at net.bible.android.view.activity.page.BibleView$ModuleStylesAssetHandler.$r8$lambda$6HMCCxaHh6-O7uIo6aOXZJ1Fhis(BibleView.kt:0)
at net.bible.android.view.activity.page.BibleView$ModuleStylesAssetHandler$$ExternalSyntheticLambda0.invoke(R8$$SyntheticClass:0)
at kotlin.text.StringsKt__AppendableKt.appendElement(Appendable.kt:84)
at kotlin.collections.CollectionsKt___CollectionsKt.joinTo(_Collections.kt:3601)
at kotlin.collections.CollectionsKt___CollectionsKt.joinToString(_Collections.kt:3618)
at kotlin.collections.CollectionsKt___CollectionsKt.joinToString$default(_Collections.kt:3617)
at net.bible.android.view.activity.page.BibleView$ModuleStylesAssetHandler.handle(BibleView.kt:939)
at androidx.webkit.WebViewAssetLoader.shouldInterceptRequest(WebViewAssetLoader.java:560)
at net.bible.android.view.activity.page.BibleView$BibleViewClient.shouldInterceptRequest(BibleView.kt:1171)
at WV.fg.a(chromium-TrichromeWebViewGoogle.aab-stable-749903430:100)
at org.chromium.android_webview.ShouldInterceptRequestMediator.shouldInterceptRequestFromNative(chromium-TrichromeWebViewGoogle.aab-stable-749903430:18)
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
at libcore.io.Linux.open(Native Method)
at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
at libcore.io.BlockGuardOs.open(BlockGuardOs.java:274)
at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8611)
at libcore.io.IoBridge.open(IoBridge.java:560)
... 14 more
```


Asked user to provide epub file.



<details><summary>Initial investigation with Claude</summary>
<p>

Root Cause

  The crash occurs at BibleView.kt:971:

  val content = styleSheets.joinToString("\n") { String(it.readBytes()) }

  The styleSheets list is built from database records in EpubBackendState.kt:150-155:

  fun styleSheets(key: Key): List<File> {
      val frag = getFragment(key)?: return emptyList()
      val file = fileForOriginalId(frag.originalId)?: return emptyList()
      val parentFolder = file.parentFile
      return dao.styleSheets(frag.originalId).map { File(parentFolder, it.styleSheetFile).canonicalFile }
  }

  The Problem

  1. The database stores stylesheet file paths referenced in the EPUB
  2. The code constructs File objects from these paths without checking if the files actually exist
  3. When readBytes() is called on a non-existent file, it throws FileNotFoundException

  This happens because the user's EPUB (Daily Walk Bible-NLT) contains references to stylesheet files (in OEBPS/Text/XXXXXXXXXXXXXXXX) that don't actually exist on disk - likely corrupted, incomplete extraction, or the EPUB has broken internal references.

  Fix

  The simplest fix is to filter out non-existent files before reading. In BibleView.kt:971, change:

  val content = styleSheets.joinToString("\n") { String(it.readBytes()) }

  To:

  val content = styleSheets.filter { it.exists() && it.canRead() }.joinToString("\n") { String(it.readBytes()) }

  Would you like me to implement this fix?

</p>
</details>

--- a/app/src/main/java/net/bible/android/view/activity/page/BibleView.kt
+++ b/app/src/main/java/net/bible/android/view/activity/page/BibleView.kt
@@ -949,7 +949,8 @@ class BibleView(val mainBibleActivity: MainBibleActivity,
                         } else null
                     }  else null) ?: return notFound
 
-                val content = styleSheets.joinToString("\n") { String(it.readBytes()) }
+                val content = styleSheets.filter() { it.exists() }
+                    .joinToString("\n") { String(it.readBytes()) }
                     .replace(colorRe, "")
                     .replace(bodyRe, "")
                     .replace(lineHeightRe, "")

